osquery/tables/system/windows/registry.cpp

```diff
@@ -266,33 +269,145 @@ void queryKey(const std::string& keyPath, QueryData& results) {
   }
   delete[](bpDataBuff);
   RegCloseKey(hRegistryHandle);
-};
+}
 
-QueryData genRegistry(QueryContext& context) {
-  QueryData results;
-  std::set<std::string> rKeys;
-  auto shouldWarnLocalUsers = false;
-  /// By default, we display all HIVEs
-  if ((context.constraints["key"].exists(EQUALS) &&
-       context.constraints["key"].getAll(EQUALS).size() > 0)) {
-    rKeys = context.constraints["key"].getAll(EQUALS);
-    shouldWarnLocalUsers = true;
-  } else {
-    for (auto& h : kRegistryHives) {
-      rKeys.insert(h.first);
+void populateDefaultKeys(std::set<std::string>& rKeys) {
```

Comment:
This is sending me internal warning alarms. Is it possible to refactor `kRegistryHives` into a set then use `std::set`'s copy-assignment constructor when you need to make a copy of the defaults?